crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+async fn get_accounts_in_wallets(
+    channel: Channel,
+    wallets: Vec<String>,
+) -> Result<(Vec<DirkAccount>, Vec<DistributedAccount>), SignerModuleError> {
+    trace!(?wallets, "Sending Lister/ListAccounts request to Dirk");
+
+    let mut client = ListerClient::new(channel);
+    let pubkeys_request = tonic::Request::new(ListAccountsRequest { paths: wallets });
+    let pubkeys_response = client
+        .list_accounts(pubkeys_request)
+        .await
+        .map_err(|err| DirkCommunicationError(format!("error listing accounts: {err}")))?;
+
+    if pubkeys_response.get_ref().state() != ResponseState::Succeeded {
+        return Err(DirkCommunicationError("list accounts request returned error".to_string()));
+    }
+
+    let inner = pubkeys_response.into_inner();
+    Ok((inner.accounts, inner.distributed_accounts))
+}
+
+pub fn aggregate_partial_signatures(
+    partials: &[BlsSignature],
+    identifiers: &[u64],
+) -> Option<BlsSignature> {
+    trace!("Aggregating partial signatures");
+    // Ensure the number of partial signatures matches the number of identifiers
+    if partials.len() != identifiers.len() {
+        warn!("aggregate_partial_signatures: Invalid number of partial signatures");
+        return None;
+    }
+
+    // Deserialize partial signatures into G2 points
+    let mut points = Vec::new();
+    for sig in partials {
+        if sig.len() != BLS_SIGNATURE_BYTES_LEN {
+            warn!("aggregate_partial_signatures: Invalid signature length");
+            return None;
+        }
+        let arr: [u8; BLS_SIGNATURE_BYTES_LEN] = (*sig).into();
+        let opt: Option<G2Affine> = G2Affine::from_compressed(&arr).into();
+        let opt: Option<G2Projective> = G2Projective::from(&opt.unwrap()).into();
+        if let Some(point) = opt {
+            points.push(point);
+        } else {
+            warn!("aggregate_partial_signatures: Failed to deserialize signature");
+            return None;
+        }
+    }
+
+    // Create a map of identifiers to their corresponding points
+    let mut shares: HashMap<u64, &G2Projective> = HashMap::new();
+    for (id, point) in identifiers.iter().zip(points.iter()) {
+        shares.insert(*id, point);
+    }
+
+    // Perform Lagrange interpolation to recover the master signature
+    let mut recovered = G2Projective::identity();
+    for (id, point) in &shares {
+        // Compute the Lagrange coefficient for this identifier
+        let mut numerator = Scalar::from(1u32);
+        let mut denominator = Scalar::from(1u32);
+        for other_id in shares.keys() {
+            if other_id != id {
+                numerator *= Scalar::from(*other_id);
+                denominator *= Scalar::from(*other_id) - Scalar::from(*id);
+            }
+        }
+        let lagrange_coeff = numerator * denominator.invert().unwrap();
+
+        // Multiply the point by the Lagrange coefficient and add to the recovered point
+        recovered += **point * lagrange_coeff;
```

Comment:
Should these be `checked` operations? Double-checking as idk

Reply:
I think it's not necessary, from what I've been reading since operations are done in a finite field there's no overflow/underflow possible, also the multiplication and addition are handled internally by the `G2Projective` struct defined in the crate.